### PR TITLE
Taylor storing image database

### PR DESCRIPTION
**Pull request by AmazingOrange**
Replaced req.body with req.session at /editUser at 2800_app.js
Cleaned up some typos at index.html.
Validated user input when user want to edit profile.

### CODE REVIEW

File: 2800_app.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const session = require("express-session");
 const sanitizeHtml = require("sanitize-html");
 const app = express();
 const fs = require("fs");
+const multer = require("multer"); // storing images
```
Comment (mksample) [inline 2800_app.js:7]: Shouldn't include half finished photo upload code.

File: 2800_app.js [hunk 2]
```diff
@@ -23,7 +24,19 @@ const remoteSqlAuthentication = {
     database: "heroku_b395e55ab1671ee"
 }
 
-const sqlAuthentication = remoteSqlAuthentication; // SETTING TO USE LOCAL OR REMOTE DB
+const sqlAuthentication = localSqlAuthentication; // SETTING TO USE LOCAL OR REMOTE DB
+
+
+// storing image at images
```
Comment (mksample) [inline 2800_app.js:30]: photo upload code should not be here

File: 2800_app.js [hunk 3]
```diff
@@ -304,6 +318,7 @@ app.post("/adminEditUser", function (req, res) {
         WHERE ID = ` + req.body.userID;
 
         con.query(editUser, function (error, results) {
+            con.end(err => {if (err) {console.log(err)}});
```
Comment (mksample) [inline 2800_app.js:321]: why was this added? this will hard crash our app if any admin tries to edit any user.

File: 2800_app.js [hunk 4]
```diff
@@ -423,6 +438,17 @@ app.post("/deleteUser", function (req, res) {
     })
 })
 
+app.post('/upload-images', upload.array("files"), function (req, res) {
```
Comment (mksample) [inline 2800_app.js:441]: photo upload code again

File: app/html/caller_profile.html [hunk 5]
```diff
@@ -90,7 +90,7 @@ <h3>Account</h3>
         <li>
           <span> <i class="material-icons">logout</i></span>
           <form action="/logout" method="get">
-            <input id="logout" type="submit" value="Logout" />
+            <input id="logginout" type="submit" value="Logout" />
```
Comment (mksample) [inline app/html/caller_profile.html:93]: why was this changed?

File: app/html/caller_profile.html [hunk 6]
```diff
@@ -191,26 +191,47 @@ <h3>Settings</h3>
     <!-- DASHBOARD CONTENT -->
     <div class="main">
       <section class="profile tabs">
+        <div id="user_profile_pictue"></div>
```
Comment (mksample) [inline app/html/caller_profile.html:194]: why was the user profile picture display removed?

File: app/html/caller_profile.html [hunk 7]
```diff
@@ -191,26 +191,47 @@ <h3>Settings</h3>
...
-          <div class="edit" id="detail_user_email" >email</div>
-          <p>Password: </p>
-          <div class="edit" id="detail_user_password" >name</div>
-          <p>Age: </p>
-          <div class="edit" id="detail_user_age" >age</div>
-          <p>Gender: </p>
-          <div class="edit" id="detail_user_gender" >gender</div>
-          <p>Phone Number: </p>
-          <div class="edit" id="detail_user_cellphone">number</div>
+          <div id="container">
```
Comment (mksample) [inline app/html/caller_profile.html:197]: photo code

File: app/html/caller_profile.html [hunk 8]
```diff
@@ -316,6 +337,40 @@ <h1>Settings</h1>
     })
   </script>
 
+  <script>
```
Comment (mksample) [inline app/html/caller_profile.html:340]: photo code

File: app/html/responder_profile.html [hunk 9]
```diff
@@ -3,12 +3,12 @@
 
 <head>
   <!-- Page Title-->
-  <title>Responder Profile Page</title>
+  <title>User ProfilePage</title>
```
Comment (mksample) [inline app/html/responder_profile.html:6]: why was this changed

File: app/html/responder_profile.html [hunk 10]
```diff
@@ -3,12 +3,12 @@
...
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
   <!-- Our StyleSheet-->
-  <link rel="stylesheet" href="/css/responder_profile.css">
+  <link rel="stylesheet" href="/css/caller_profile.css">
```
Comment (mksample) [inline app/html/responder_profile.html:11]: why does the responder profile page now use caller profile styling

File: app/html/responder_profile.html [hunk 11]
```diff
@@ -188,25 +190,34 @@ <h3>Settings</h3>
...
+          <div class="edit" id="detail_user_password" >name</div>
+          <p>Age: </p>
+          <div class="edit" id="detail_user_age" >age</div>
+          <p>Gender: </p>
+          <div class="edit" id="detail_user_gender" >gender</div>
+          <p>Phone Number: </p>
+          <div class="edit" id="detail_user_cellphone">number</div>
+          <p>Role: </p>
+          <div class="edit" id="detail_user_role">role</div>
+          <br><br>
```
Comment (mksample) [inline app/html/responder_profile.html:212]: responder profile page doesn't show editing errors?

File: app/html/xresponder_profile.html [hunk 12]
```diff
@@ -0,0 +1,310 @@
+<!DOCTYPE html>
```
Comment (mksample) [inline app/html/xresponder_profile.html:1]: what is this file

File: public/css/caller_profile.css [hunk 13]
```diff
@@ -429,6 +429,10 @@ aside .search-bar button {
     outline: none;
 }
 
+#user_profile_image{
```
Comment (mksample) [inline public/css/caller_profile.css:432]: this ID doesn't exist in the html file

File: public/js/responder_profile_edit.js [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+        firstName: document.getElementById("detail_user_firstN").textContent,
+        lastName: document.getElementById("detail_user_lastN").textContent,
+        age: document.getElementById("detail_user_age").textContent,
+        gender: document.getElementById("detail_user_gender").textContent,
+        phoneNumber: document.getElementById("detail_user_cellphone").textContent,
+        role: document.getElementById("detail_user_role").textContent
+    });
+
+    console.log(input); 
+    xhttp.send(input);
```
Comment (mksample) [inline public/js/responder_profile_edit.js:77]: responder profile editing doesn't check for errors?

File: public/js/responder_profile_edit.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
+"use strict";
+ready(async function () {
```
Comment (mksample) [inline public/js/responder_profile_edit.js:2]: function not used or referenced in file

**Review by mksample — CHANGES_REQUESTED** [review]
Needs a bit of work, also not sure why profile pictures in imgs/userProfile/ were added.

File: app/html/caller_profile.html (hunk 5 above)
Comment (AmazingOrange) [reply app/html/caller_profile.html:93]: this id is repeating.

File: 2800_app.js (hunk 4 above)
Comment (AmazingOrange) [reply 2800_app.js]: Removed uncompleted codes.

File: 2800_app.js (hunk 3 above)
Comment (AmazingOrange) [reply 2800_app.js]: Removed this code.

File: 2800_app.js (hunk 2 above)
Comment (AmazingOrange) [reply 2800_app.js]: Removed uncompleted codes.

File: 2800_app.js (hunk 1 above)
Comment (AmazingOrange) [reply 2800_app.js]: Removed uncompleted codes.

File: app/html/responder_profile.html (hunk 11 above)
Comment (AmazingOrange) [reply app/html/responder_profile.html:212]: Right now, I have responder_profile.html validated the user input.

File: app/html/xresponder_profile.html (hunk 12 above)
Comment (AmazingOrange) [reply app/html/xresponder_profile.html:1]: xresponder_profile.html is removed.

File: public/css/caller_profile.css (hunk 13 above)
Comment (AmazingOrange) [reply public/css/caller_profile.css:432]: Oh, thank you for pointing it out. Forgot to delete.

**Review by mksample — APPROVED** [review]
